client/catch-js-errors.js

```diff
@@ -0,0 +1,76 @@
+( function() {
+	var savedWindowOnError = window.onerror;
+
+	// Props http://stackoverflow.com/a/20405830/379063
+	function stringifyErrorForUrlParams( error ) {
+		var simpleObject = {};
+		Object.getOwnPropertyNames( error ).forEach( function( key ) {
+			simpleObject[ key ] = encodeURIComponent( error[ key ] );
+		} );
+		return JSON.stringify( simpleObject );
+	};
+
+	// Props http://stackoverflow.com/a/17604754/379063
+	function isLocalStorageNameSupported() {
+		var testKey = 'test',
+			storage = window.localStorage;
+		try {
+			storage.setItem( testKey, '1' );
+			storage.removeItem( testKey );
+			return true;
+		} catch ( error ) {
+			return false;
+		}
+	}
+
+	/**
+	 * Throttle function by delay ms
+	 *
+	 * @param {function} fn wrapped function
+	 * @param {number} delay minimum debounce time in ms
+	 * @returns {function} a callback
+	 */
+	function debounce( fn, delay ) {
+		var lastCall = Date.now();
+
+		return function() {
+			var now = Date.now();
+
+			if ( ( now - lastCall ) < delay ) {
+				return;
```

Comment:
probably want to stash the error we would have logged in an array or stack so we don't lose the first error in a sequence.